src/sentry/static/sentry/app/views/settings/incidentRules/triggers/form.tsx

```diff
@@ -34,97 +32,130 @@ type Props = {
    */
   error?: {[fieldName: string]: string};
   projects: Project[];
+  resolveThreshold: UnsavedIncidentRule['resolveThreshold'];
+  thresholdType: UnsavedIncidentRule['thresholdType'];
   trigger: Trigger;
   triggerIndex: number;
   isCritical: boolean;
 
   onChange: (trigger: Trigger, changeObj: Partial<Trigger>) => void;
+  onThresholdTypeChange: (thresholdType: AlertRuleThresholdType) => void;
 };
 
+const CriticalIndicator = styled(CircleIndicator)`
+  background: ${p => p.theme.red400};
+  margin-right: ${space(1)};
+`;
+
+const WarningIndicator = styled(CircleIndicator)`
+  background: ${p => p.theme.yellow400};
+  margin-right: ${space(1)};
+`;
+
+const ResolvedIndicator = styled(CircleIndicator)`
+  background: ${p => p.theme.green400};
+  margin-right: ${space(1)};
+`;
+
 class TriggerForm extends React.PureComponent<Props> {
-  getThresholdKey = (
-    type: AlertRuleThreshold
-  ): AlertRuleThresholdKey[AlertRuleThreshold] =>
-    type === AlertRuleThreshold.RESOLUTION ? 'resolveThreshold' : 'alertThreshold';
+  // Overwritten in ResolvedTriggerForm
+  getThresholdType = () => {
+    return this.props.thresholdType;
+  };
+
+  getTriggerLabel = (): React.ReactNode => {
+    const {isCritical} = this.props;
+    const triggerLabel = isCritical ? t('Critical Status') : t('Warning Status');
+    const TriggerIndicator = isCritical ? CriticalIndicator : WarningIndicator;
+    return (
+      <React.Fragment>
+        <TriggerIndicator size={12} />
+        {triggerLabel}
+      </React.Fragment>
+    );
+  };
+
+  getFieldHelp = (): React.ReactNode => {
+    const {isCritical} = this.props;
+    return tct('The threshold that will activate the [severity] status.', {
+      severity: isCritical ? t('critical') : t('warning'),
+    });
+  };
```

Comment:
Instead of OOP here, the "react way" of doing things is to have theses as props, so in the parent where you render these components, you pass in the data.

Reply:
that makes way more sense